elyra-ai/elyra
# Migrate to the latest version of canvas

Migrate to the latest canvas version

- [x] Upgrade `@elyra/canvas` (from **6.1.23** to **8.0.32**)
- [x] Resolve breaking changes
- [x] Clean-up the UI
- [x] Verify ui tests
- [x] Update documentation screenshots
- Confirm canvas upgrade fixes issues:
    - ~~#750 ([part 2](https://github.com/elyra-ai/elyra/issues/750#issuecomment-657894923))~~ // will be resolved in canvas 9.0.x
    - [x] [icon svg support](https://github.com/elyra-ai/elyra/issues/733#issuecomment-656345530)
    - [x] #386
- Merge (upcoming) Pipeline editor PRs: 
    - [x] #741
    - ~~#744~~ // will be merged after this PR
    - [x] #750 ([part 1](https://github.com/elyra-ai/elyra/issues/750#issuecomment-657894923))
    - ~~#752~~ // may be merged after this PR

Changes

- per @elyra/canvas release 7.0.5
    - Updated dependency `react-intl` (^3.0.0) 
    - Added dependency `react-redux` (^7.0.0)
- per @elyra/canvas release 8.0.1
    - Updated CSS class names and styling
    - Placed `<CommonCanvas>` inside a `<IntlProvider>`
    - Removed `toolbarMenuActionHandler`. Handling of these actions moved to the `editActionHandler`
    - Removed `contextMenuActionHandler`. Handling of these actions moved to the `editActionHandler`.
    - Renamed `addComment` action to `createAutoComment`
    - Renamed `delete` action to `deleteSelectedObjects`
- per @elyra/canvas release 8.0.2
    - Added config `createSupernode` // user can disable creation of supernodes via configuration parameter
- per @elyra/canvas release 8.0.4
    - Added config `enableInsertNodeDroppedOnLink` // user can drag nodes from the palette and drop them onto existing links in the flow which causes the node to be inserted between the two nodes joined by the link

- Update [screenshots in documentation](https://elyra.readthedocs.io/en/latest/user_guide/pipelines.html)

Fixes #645
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.
